python_modules/dagster/dagster_tests/core_tests/test_object_store.py

```diff
@@ -123,3 +152,49 @@ def test_s3_object_store_with_custom_serializer():
         assert object_store.get_object(context, LowercaseString.inst(), ['foo']) == 'foo'
     finally:
         object_store.rm_object(context, ['foo'])
+
+
+def test_file_system_object_store_with_type_storage_plugin():
+    run_id = str(uuid.uuid4())
+
+    # FIXME need a dedicated test bucket
+    object_store = FileSystemObjectStore(
+        run_id=run_id, types_to_register={String.inst(): FancyStringFilesystemTypeStoragePlugin}
+    )
+
+    try:
+        with yield_pipeline_execution_context(
+            PipelineDefinition([]), {}, RunConfig(run_id=run_id)
+        ) as context:
+            object_store.set_object('hello', context, String.inst(), ['obj_name'])
+
+        assert object_store.has_object(context, ['obj_name'])
```

Comment:
all interactions with object_store should be within the with block. I think we should add an explicit closed state to the pipeline_execution_context to catch this type of stuff. Curious what @alangenfeld thinks

Reply:
I like that idea, it's very tempting esp in tests to do it this way

Reply:
its not obvious to me the best way to do it, but I do think we want to try to have some notion of closed state to prevent issues from only being surfaced when the torn down resource gets accessed